File: app/src/main/java/com/codekk/mvp/presenter/JobListPresenterImpl.java
package com.codekk.mvp.presenter;

import com.codekk.mvp.model.JobListModel;
import com.codekk.mvp.view.ViewManager;
import com.codekk.net.Api;
import com.codekk.net.NetFunc;
import com.codekk.ui.base.BasePresenterImpl;

import io.reactivex.network.RxNetWork;


/**
 * by y on 2017/5/18.
 */

public class JobListPresenterImpl extends BasePresenterImpl<ViewManager.JobListView, JobListModel> implements PresenterManager.JobListPresenter {

    public JobListPresenterImpl(ViewManager.JobListView view) {
        super(view);
    }

    @Override
    public void netWorkRequest(int page) {
        netWork(RxNetWork
                .observable(Api.JobService.class)
                .getJobList(page)
                .map(new NetFunc<>()));
    }

    @Override
    public void onNetWorkSuccess(JobListModel data) {
        if (data.getSummaryArray().isEmpty()) {
            view.noMore();
        } else {
            super.onNetWorkSuccess(data);
        }
    }
}

File: app/src/main/java/com/codekk/mvp/presenter/JobListPresenterImpl.kt
package com.codekk.mvp.presenter

import com.codekk.JobService
import com.codekk.NetFunc
import com.codekk.mvp.model.JobListModel
import com.codekk.mvp.view.ViewManager
import com.codekk.ui.base.BasePresenterImpl

import io.reactivex.network.RxNetWork


/**
 * by y on 2017/5/18.
 */

class JobListPresenterImpl(view: ViewManager.JobListView) : BasePresenterImpl<ViewManager.JobListView, JobListModel>(view), PresenterManager.JobListPresenter {

    override fun netWorkRequest(page: Int) {
        netWork(RxNetWork
                .observable(JobService::class.java)
                .getJobList(page)
                .map(NetFunc()))
    }

    override fun onNetWorkSuccess(data: JobListModel) {
        if (data.summaryArray.isEmpty()) {
            view?.noMore()
        } else {
            super.onNetWorkSuccess(data)
        }
    }
}
